modules/wpaperd/hm.nix

```
@@ -9,9 +9,19 @@
       mode = let
         inherit (config.stylix) imageScalingMode;
       in
-        if imageScalingMode == "fill"
+        if imageScalingMode == "stretch"
+        then "stretch"
+        else if imageScalingMode == "fill"
+        # wpaperd's center mode is closest to the described behavior of fill
+        then "center"
+        else if imageScalingMode == "fit"
         then "fit"
-        else imageScalingMode;
+        else if imageScalingMode == "tile"
+        then "tile"
+        else if imageScalingMode == "center"
+        # wpaperd doesn't have any mode close to the described behavior of center
+        then null
```

Would the following improve the UX or not:

```suggestion
        then
          lib.info
          "[stylix] wpaperd has not equivalent center mode for scaling the image"
          null
```

---

Added the warning. Actually, `formats.toml` was complaining when there was a null value, so I changed the structure around a bit. I'm not super experienced with nix so lmk if there's a better way to do it.

---

> Actually, `formats.toml` was complaining when there was a null value

What file are you referring to? Is this in your setup?

> I changed the structure around a bit. I'm not super experienced with nix so lmk if there's a better way to do it.

Since we do not have an abstraction layer on top of `stylix.imageScalingMode`, we might want to keep it simple with a series of if-else expression, as before.

---

When I set `mode = null;` I get `A definition for option `home-manager.users.hazel.programs.wpaperd.settings.any.mode' is not of type `TOML value'`. So I think it needs to be unset rather than null which is why I changed it. I can make it a series of if-else expressions, but I'm not sure how else to do it besides merging the attrsets.

---

Alright. The current approach sounds good to me then.